assets/app/scripts/filters/resources.js

```diff
@@ -845,4 +845,38 @@ angular.module('openshiftConsole')
 
       return portDisplayValue(servicePort.port, servicePort.targetPort, servicePort.protocol);
     };
+  })
+  .filter('routeCondition', function() {
+    return function(route, type) {
+      if (!route.status.ingress) {
+        return null;
+      }
+      var found = null;
+      angular.forEach(route.status.ingress, function(ingress) {
+        angular.forEach(ingress.conditions, function(condition) {
+          if (condition.type === type) {
+            found = condition;
+          }
+        });
+      });
+      return found;
+    };
+  })
+  .filter('routeHost', function() {
+    return function (route) {
+      if (!route.status.ingress || route.status.ingress.length === 0) {
```

Comment:
@smarterclayton is route.status.ingress null, or will it just be an empty array, or could it be either